course/test/src/Ant_TEST.cc

```diff
@@ -0,0 +1,129 @@
+#include <cmath>
+#include <sstream>
+#include <string>
+#include <memory>
+
+#include "gtest/gtest.h"
+
+#include "ant.h"
+#include "cell_mock.h"
+
+namespace ekumen {
+namespace simulation {
+namespace test {
+namespace {
+
+GTEST_TEST(AntTest, OccupiedCellOnInitialization) {
+  std::shared_ptr<Cell> cell(new Cell());
+  Ant ant(cell);
+  EXPECT_FALSE(cell->IsFree());
+}
+
+GTEST_TEST(AntTest, CallbackAssignedCorrectly) {
+  std::shared_ptr<Cell> cell(new Cell());
+  Ant ant = Ant(cell);
+  bool lambda_called = false;
+  auto callback = [&](InsectCallbackMetrics& metrics) {
+        lambda_called = true;
+  };
+  ant.SetRoundResultsCallback(callback);
+  ant.RunRound();
+
+  EXPECT_TRUE(lambda_called);
+}
+
+GTEST_TEST(AntTest, IsAliveOnInit) {
+  std::shared_ptr<Cell> cell(new Cell());
+  Ant ant = Ant(cell);
+  InsectCallbackMetrics test_metrics;
```

Comment:
You don't really need this structure for this test, right?

Reply:
Actually, I defined that structure as part of the class Insect.

Recall that the only way of insects to communicate to other classes what happened after a simulation round is through a callback. This callback receives a reference to a InsectCallbackMetrics instance.